core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -66,6 +67,12 @@ private void verifyInputAndOutputFiles(Set<DataFile> dataFilesToDelete, Set<Dele
     }
   }
 
+  @Override
+  public RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> filesToAdd, long sequenceNumber) {
```

Comment:
Slightly unrelated concern: should we be using `Set` here? It seems needlessly restrictive. Plus, `DataFile` is an interface, so you could easily pass files that don't implement equals/hashSet and are always considered unique. Just one thing that's always made me wonder about this API.

Not a blocker for this PR though!